### PR TITLE
chore: fixed workflow go version

### PR DESCRIPTION
**Pull request by Two-Hearts**
Always check-latest go version in workflows.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,6 +34,8 @@ jobs:
         uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
         with:
           go-version: ${{ matrix.go-version }}
+          cache: false
```
Comment (shizhMSFT) [inline .github/workflows/build.yml:37]: We still can use the cache as long as it is the latest.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**